### PR TITLE
DCS-1582 actor service hierarchy

### PR DESCRIPTION
**Pull request by rafaelgarrote**

### CODE REVIEW

File: core/pom.xml [hunk 1]
```diff
@@ -38,6 +38,13 @@
             <type>test-jar</type>
             <scope>test</scope>
         </dependency>
+        <dependency>
```
Comment (fhuertas) [inline core/pom.xml]: You can use the version of parent pom

File: core/src/test/scala/com/stratio/tikitakka/core/UpAndDownActorTest.scala [hunk 2]
```diff
@@ -12,7 +12,7 @@ import com.stratio.tikitakka.common.model.ContainerInfo
 import com.stratio.tikitakka.updown.DummyUpAndDownComponent
```
Comment (fhuertas) [inline core/src/test/scala/com/stratio/tikitakka/core/UpAndDownActorTest.scala]: Should the test name end with Test or UnitTest?

File: discovery/src/test/scala/com/stratio/tikitakka/columbus/consul/ConsulComponentIT.scala [hunk 3]
```diff
@@ -1,18 +1,3 @@
-/*
```
Comment (fhuertas) [inline discovery/src/test/scala/com/stratio/tikitakka/columbus/consul/ConsulComponentIT.scala]: Why you remote the licence?

File: pom.xml [hunk 4]
```diff
@@ -44,6 +44,7 @@
         <tikitakka.common.version>0.1.0-SNAPSHOT</tikitakka.common.version>
         <tikitakka.discovery.version>0.1.0-SNAPSHOT</tikitakka.discovery.version>
         <tikitakka.orchestration.version>0.1.0-SNAPSHOT</tikitakka.orchestration.version>
+        <tikitakka.common.version>0.1.0-SNAPSHOT</tikitakka.common.version>
```
Comment (fhuertas) [inline pom.xml]: This is duplicated with the line 44

File: core/src/main/scala/com/stratio/tikitakka/core/Dependencies.scala [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+import com.stratio.tikitakka.columbus.DiscoveryComponent
+
+trait Dependencies {
+
+  //Actor System
+  val actorSystem: ActorSystem
+  val materializer: ActorMaterializer
+
+  //Actors
+  val discoveryActor: DiscoveryActor
```
Comment (dvallejo) [inline core/src/main/scala/com/stratio/tikitakka/core/Dependencies.scala]: Where do you need an Actor type? I think we always use ActorRef.

File: core/src/main/scala/com/stratio/tikitakka/core/ServicesActor.scala [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+
+  var services = Map[String, Cancellable]()
+  lazy val discoverInitialDelayInSeconds = ConfigComponent.getInt("discover.initial.delay.in.seconds", 5) second
+  lazy val discoverIntervalInSeconds = ConfigComponent.getInt("discover.interval.in.seconds", 5) second
+
+  override def receive: Receive = {
+
+    case AppsDiscovered(appsDiscovered) => appsDiscovered.foreach{ app =>
+      services.get(app._1) match {
+        case Some(_) => log.debug(s"[${app._1}] already managed")
```
Comment (dvallejo) [inline core/src/main/scala/com/stratio/tikitakka/core/ServicesActor.scala]: In this foreach you only need app._1. For this reason you can extract only this part:
```scala
appsDiscovered.foreach { case (appId, _) =>
  //....
}
```

File: discovery/src/main/scala/com/stratio/tikitakka/columbus/consul/ConsulComponent.scala [hunk 7]
```diff
@@ -33,6 +50,21 @@ trait ConsulComponent extends DiscoveryComponent with HttpRequestUtils with LogU
     )
   }
 
+  def discover(serviceName: String): Future[Option[DiscoveryAppInfo]] = {
+    discoverServiceRequest(serviceName) map ( response =>
+      response match {
+        case x :: _ =>
+          Some(DiscoveryAppInfo(
+            response.head.ID.getOrElse(response.head.ServiceName), //Future versions will have ID
```
Comment (dvallejo) [inline discovery/src/main/scala/com/stratio/tikitakka/columbus/consul/ConsulComponent.scala]: Here, you can also implement this method using headOption:
```scala
discoverServiceRequest(serviceName)
  .map { response =>
    response.headOption.map { head =>
      DiscoveryAppInfo(
        head.ID.getOrElse(response.head.ServiceName),
        head.ServiceName,
        response.map(_.toDiscoveryServiceInfo),
        response.map(_.ServiceTags).reduce(_ ++ _)
      )
    }
  }
```

**Pull request merged**